tagstudio/src/qt/ts_qt.py

```diff
@@ -902,6 +925,141 @@ def add_tags_to_selected_callback(self, tag_ids: list[int]):
         for entry_id in self.selected:
             self.lib.add_tags_to_entry(entry_id, tag_ids)
 
+    def delete_files_callback(self, origin_path: str | Path, origin_id: int | None = None):
+        """Callback to send on or more files to the system trash.
+
+        If 0-1 items are currently selected, the origin_path is used to delete the file
+        from the originating context menu item.
+        If there are currently multiple items selected,
+        then the selection buffer is used to determine the files to be deleted.
+
+        Args:
+            origin_path(str): The file path associated with the widget making the call.
+                May or may not be the file targeted, depending on the selection rules.
+            origin_id(id): The entry ID associated with the widget making the call.
+        """
+        entry: Entry | None = None
+        pending: list[tuple[int, Path]] = []
+        deleted_count: int = 0
+
+        if len(self.selected) <= 1 and origin_path:
+            origin_id_ = origin_id
+            if not origin_id_:
+                with contextlib.suppress(IndexError):
+                    origin_id_ = self.selected[0]
+
+            pending.append((origin_id_, Path(origin_path)))
+        elif (len(self.selected) > 1) or (len(self.selected) <= 1 and not origin_path):
```

Comment:
Nit: this can be simplified because the previous condition was false
```suggestion
        elif (len(self.selected) > 1) or (len(self.selected) <= 1):
```